src/event_handler.rs

```diff
@@ -192,22 +188,125 @@ async fn send_execution_output(app_state: Data<AppState>, mut rx: Receiver<JobRe
                     job_output.execution_response.total_time.into(),
                     job_output.execution_response.error_code.into(),
                     job_output.sign_timestamp,
+                ),
+            );
+            resend_deadline = (app_state.execution_buffer_time as u128) + job_output.user_deadline
+                - job_output.execution_response.total_time;
+            txn_type = "output";
+            job_id = job_output.id;
+        } else if job_response.timeout_response.is_some() {
+            // Assign details of job execution timeout transaction
+            job_id = job_response.timeout_response.unwrap();
+            txn = Some(
+                Jobs::new(
+                    app_state.jobs_contract_addr,
+                    app_state.http_rpc_client.lock().unwrap().clone().unwrap(),
+                )
+                .slash_on_execution_timeout(job_id),
+            );
+        }
+
+        if txn.is_none() {
+            continue;
+        }
+
+        let mut update_nonce = false;
+        let job_txn = txn.unwrap();
+
+        // Retry loop for sending a transaction to the common chain
+        for _retry in 0..3 {
+            let txn = job_txn.clone();
+
+            // Estimate gas required for the transaction to execute and retry otherwise
+            let estimated_gas = txn.estimate_gas().await;
+            let Ok(estimated_gas) = estimated_gas else {
+                eprintln!("Failed to estimate gas from the rpc for sending execution {} transaction for job id {}: {:?}", txn_type, job_id, estimated_gas.unwrap_err());
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
+
+            // Get current gas price for the common chain network and retry otherwise
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let gas_price = http_rpc_client.get_gas_price().await;
+            let Ok(gas_price) = gas_price else {
```

Comment:
what was the reason to use gas price? let the provider handle it.

Reply:
When sending a replacement transaction for a nonce , the network requires it to have a higher gas price (typically more than 10%) than the original transaction in the mempool. So to keep track of what gas price to send with the replacement transaction I have calculated this beforehand so that "replacement transaction underpriced" error is sure to not pop up while resending.

Reply:
can we get gas_price from pending transaction? there should be some record for the gas_price used.

Reply:
No the pending transaction object doesn't store the gas price used, that's why had to calculate and store it separately.